pinot-core/src/test/java/com/linkedin/pinot/core/segment/index/loader/SegmentPreProcessorTest.java

```diff
@@ -31,143 +32,152 @@
-        .createFromLocalFS(segmentDirectoryFile, segmentMetadata, ReadMode.mmap)) {
-      try (SegmentDirectory.Reader reader = segmentDirectory.createReader()) {
-        Assert.assertFalse(reader.hasIndexFor(column1Name, ColumnIndexType.INVERTED_INDEX));
-        Assert.assertFalse(reader.hasIndexFor(column13Name, ColumnIndexType.INVERTED_INDEX));
-        // already created by segment gen specifying config
-        Assert.assertTrue(reader.hasIndexFor(column7Name, ColumnIndexType.INVERTED_INDEX));
-        Assert.assertFalse(reader.hasIndexFor(noSuchColumnName, ColumnIndexType.INVERTED_INDEX));
-      }
+    String col7FileName = segmentMetadata.getBitmapInvertedIndexFileName(COLUMN7_NAME, segmentVersion);
+    File col7File = new File(_segmentDirectoryFile, col7FileName);
+    Assert.assertTrue(col7File.exists());
+    FileTime preLastModifiedTime = Files.getLastModifiedTime(col7File.toPath());
+
+    checkInvertedIndexCreation(_segmentDirectoryFile, segmentMetadata);
+
+    // check we didn't create column7 index again
+    Assert.assertEquals(Files.getLastModifiedTime(col7File.toPath()), preLastModifiedTime);
+  }
+
+  @Test
+  public void testV3Process()
+      throws Exception {
+    SegmentV1V2ToV3FormatConverter converter = new SegmentV1V2ToV3FormatConverter();
+    converter.convert(_segmentDirectoryFile);
+    File v3SegmentDirectoryFile = new File(_segmentDirectoryFile, V3_SEGMENT_NAME);
 
-      SegmentPreProcessor processor =
-          new SegmentPreProcessor(segmentDirectoryFile, segmentMetadata, indexLoadingConfigMetadata);
-      processor.process();
-
-      long newLength = 0;
-      try (SegmentDirectory.Reader reader = segmentDirectory.createReader()) {
-        // new II added; existing not removed
-        Assert.assertTrue(reader.hasIndexFor(column1Name, ColumnIndexType.INVERTED_INDEX));
-        Assert.assertTrue(reader.hasIndexFor(column13Name, ColumnIndexType.INVERTED_INDEX));
-        Assert.assertTrue(reader.hasIndexFor(column7Name, ColumnIndexType.INVERTED_INDEX));
-        Assert.assertFalse(reader.hasIndexFor(noSuchColumnName, ColumnIndexType.INVERTED_INDEX));
-
-        try (PinotDataBuffer col1Buffer = reader.getIndexFor(column1Name, ColumnIndexType.INVERTED_INDEX)) {
-          newLength += col1Buffer.size();
-        }
-        try (PinotDataBuffer col13Buffer = reader.getIndexFor(column13Name, ColumnIndexType.INVERTED_INDEX)){
-          newLength += col13Buffer.size();
-        }
+    SegmentMetadataImpl segmentMetadata = new SegmentMetadataImpl(v3SegmentDirectoryFile);
+    String segmentVersion = segmentMetadata.getVersion();
+    Assert.assertEquals(SegmentVersion.valueOf(segmentVersion), SegmentVersion.v3);
+
+    File singleFileIndex = new File(v3SegmentDirectoryFile, "columns.psf");
+    long singleFileSize = singleFileIndex.length();
+
+    checkInvertedIndexCreation(v3SegmentDirectoryFile, segmentMetadata);
+
+    long addedLength = 0L;
```

Comment:
For v1 and v3, process second time to see that inverted indexes are not created again